src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.IncludeElementExpander.cs

```diff
@@ -220,12 +220,15 @@ private XNode[] Rewrite(XNode node, string currentXmlFilePath, CSharpSyntaxNode
                             if (ElementNameIs(element, DocumentationCommentXmlNames.ParameterElementName) ||
                                 ElementNameIs(element, DocumentationCommentXmlNames.ParameterReferenceElementName))
                             {
-                                BindName(attribute, originatingSyntax, isParameter: true);
+                                BindName(attribute, originatingSyntax, isParameter: true, isTypeRef: false);
                             }
-                            else if (ElementNameIs(element, DocumentationCommentXmlNames.TypeParameterElementName) ||
-                                ElementNameIs(element, DocumentationCommentXmlNames.TypeParameterReferenceElementName))
+                            else if (ElementNameIs(element, DocumentationCommentXmlNames.TypeParameterElementName))
                             {
-                                BindName(attribute, originatingSyntax, isParameter: false);
+                                BindName(attribute, originatingSyntax, isParameter: false, isTypeRef: false);
+                            }
+                            else if(ElementNameIs(element, DocumentationCommentXmlNames.TypeParameterReferenceElementName))
+                            {
+                                BindName(attribute, originatingSyntax, isParameter: false, isTypeRef: true);
```

Comment:
>isTypeRef [](start = 91, length = 9)

nit: consider renaming `isTypeRef` to `isTypeParameterRef` (here and throughout the stack)

Reply:
We special case the included files, meaning the logic between a inline and an included document is actually different. Primarily the inline case uses the binder factory and visitor to do binding, whereas inline has its own logic (the binder factory uses parent nodes to figure out the relationships, and in the include case we're in a different syntax tree, so the parent relationship isn't there). 

I played around trying to make the factory work for the include case, but it started getting quite complicated; it seemed simpler to fix the 'special' include binding logic to properly handle this one extra case, rather than swap in the more generic walker.